services/gitdiff/gitdiff.go

```diff
@@ -1235,8 +1235,14 @@ func SyncAndGetUserSpecificDiff(ctx context.Context, userID int64, pull *issues_
 	}
 
 	changedFiles, err := gitRepo.GetFilesChangedBetween(review.CommitSHA, latestCommit)
+	// There are way too many possible errors.
+	// Examples are various git errors such as the commit the review was based on was gc'ed and hence doesn't exist anymore as well as unrecoverable errors where we should serve a 500 response
+	// Due to the current architecture and physical limitation of needing to compare explicit error messages, we can only choose one approach without the code getting ugly
+	// For SOME of the errors such as the gc'ed commit, it would be best to mark all files as changed
+	// But as that does not work for all potential errors, we simply mark all files as unchanged and drop the error which always works, even if not as good as possible
 	if err != nil {
-		return diff, err
+		changedFiles = []string{}
```

Comment:
Alternatively,
```suggestion
		changedFiles = nil
```
or even
```suggestion
```
would be possible. Which approach is preferred?
I chose the current one due to being a little bit more explicit about its intention

Reply:
`changedFiles = nil` is clearer to me. 

But  I  think you can also just leave the `changedFiles` as it is and no need to change it.

Reply:
I'm fine with all approaches.
That's why I offered all three approaches.

> But I think you can also just leave the changedFiles as it is and no need to change it.

Do you mean
```diff
-		changedFiles = []string{}
```
or as it is right now?

Reply:
Yup, if you could agree, we can take it to try